data_models/data_refinery_models/models.py

```diff
@@ -17,26 +18,101 @@ class Meta:
         abstract = True
 
 
-# This model still has a prototypical status, but I needed something to
-# test with and it's at least in the right ballpark
+class SurveyJob(TimeTrackedModel):
+    source_type = models.CharField(max_length=256)
+
+    # The start time of the query used to replicate
+    replication_started_at = models.DateTimeField(null=True)
+
+    # The end time of the query used to replicate
+    replication_ended_at = models.DateTimeField(null=True)
+
+    # The start time of the job
+    start_time = models.DateTimeField(null=True)
+
+    # The end time of the job
+    end_time = models.DateTimeField(null=True)
+
+    class Meta:
+        db_table = "survey_jobs"
+
+
+class BatchStatuses(Enum):
+    NEW = "NEW"
+    DOWNLOADING = "DOWNLOADING"
+    DOWNLOADED = "DOWNLOADED"
+    PROCESSING = "PROCESSING"
+    PROCESSED = "PROCESSED"
+
+
 class Batch(TimeTrackedModel):
+    survey_job = models.ForeignKey(SurveyJob)
     source_type = models.CharField(max_length=256)
     size_in_bytes = models.IntegerField()
     download_url = models.CharField(max_length=2048)
-    raw_format = models.CharField(max_length=256)
-    processed_format = models.CharField(max_length=256)
-    processor_required = models.IntegerField()
-    accession_code = models.CharField(max_length=256)
+    raw_format = models.CharField(max_length=256, null=True)
+    processed_format = models.CharField(max_length=256, null=True)
+    pipeline_required = models.CharField(max_length=256)
+    accession_code = models.CharField(max_length=32)
```

Comment:
Hmm, SRA looks to be ~10 (also appears to be incrementing):
https://www.ncbi.nlm.nih.gov/sra/?term=human

GEO also appears to increment so let's say roughly 10.

The most recent ones in arrayexpress seem to be ~10:
https://www.ebi.ac.uk/arrayexpress/search.html?query=human&sortby=releasedate&sortorder=descending

ENA also appears to be incrementing with ~750k samples:
http://www.ebi.ac.uk/ena/data/search?query=human

I guess 32 characters should be enough for anybody (at least for a while?). Such statements are never proven wrong by history ;)